### ISSUE TITLE
Extra whitespace breaking babel-plugin-jsx-dom-expressions

### ISSUE THREAD

**Issue by Akkuma** [ryansolid/solid#189]
Code I'm writing:
```js
import { createResource } from 'solid-js';
import { For } from 'solid-js/dom';

export function AudioOutputSelector(props) {
  return (
    <select>
      <For each={props.devices} fallback={<div>test</div>}>
        {_ => <option value={_.deviceId}>{_.label}</option>} 
      </For>    
    </select>
  );
}
```
```
[!] (plugin babel) TypeError: /Users/omni/Documents/git/Akk/components/AudioOutputSelector.js: Cannot read property 'text' of null
components/AudioOutputSelector.js
TypeError: /Users/omni/Documents/git/Akk/components/AudioOutputSelector.js: Cannot read property 'text' of null
    at /Users/omni/Documents/git/Akk/node_modules/babel-plugin-jsx-dom-expressions/index.js:996:19
    at Array.reduce (<anonymous>)
    at transformChildren (/Users/omni/Documents/git/Akk/node_modules/babel-plugin-jsx-dom-expressions/index.js:994:8)
    at transformElement (/Users/omni/Documents/git/Akk/node_modules/babel-plugin-jsx-dom-expressions/index.js:631:5)
    at transformNode (/Users/omni/Documents/git/Akk/node_modules/babel-plugin-jsx-dom-expressions/index.js:1819:12)
    at PluginPass.transformJSX (/Users/omni/Documents/git/Akk/node_modules/babel-plugin-jsx-dom-expressions/index.js:1801:18)
    at newFn (/Users/omni/Documents/git/Akk/node_modules/@babel/core/node_modules/@babel/traverse/lib/visitors.js:179:21)
    at NodePath._call (/Users/omni/Documents/git/Akk/node_modules/@babel/core/node_modules/@babel/traverse/lib/path/context.js:55:20)
    at NodePath.call (/Users/omni/Documents/git/Akk/node_modules/@babel/core/node_modules/@babel/traverse/lib/path/context.js:42:17)
    at NodePath.visit (/Users/omni/Documents/git/Akk/node_modules/@babel/core/node_modules/@babel/traverse/lib/path/context.js:90:31)
```

Edit: Updated title to reference real issue (original: babel-jsx-dom breaking w/ For)

**Comment by ryansolid** [comment]
This one was tricky for me to reproduce since I use Prettier in most places. This caused by the extra space after the closing `For` tag before the close of the `select` tag. So clearly there is a bug in white space trimming/processing logic. It is supposed to leave a single space but I'm gathering something isn't expecting it there. I will work to get this cleared up. For now remove that trailing white space.

**Comment by Akkuma** [comment]
Thanks for the catch. I didn't even realize I had extra spaces, so I definitely need to get prettier up and going again.

**Comment by ryansolid** [comment]
Ok.. this is fixed in Babel Plugin JSX DOM Expressions 0.19.7. You will probably need to remove your node_modules blow away your package-lock to have this install properly (as Babel Preset Solid will settle for 0.19.6 if it's there). I will update Babel Preset Solid to set it as it's min version with the next Solid patch release.

**Comment by ryansolid** [comment]
Updated in Solid preset dep now. Closing.
